pkg/osutil/pool.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package osutil
```

Comment:
is there a 3rd thread pool? maybe we can use it directly, no need to reinvent the wheel.

Reply:
Actually, I try to find a 3rd pool and not found a suitable one. This pool implementation is reference to https://github.com/valyala/fasthttp/blob/master/workerpool.go#L16. It is lightweight, have self-tuning size.

Reply:
you must include the project's license here.